gips/0065-subgraph-availability-manager.md

```diff
@@ -0,0 +1,127 @@
+
+The `executionThreshold` will be set to three during contract construction. Whenever an oracle casts a vote, the `SubgraphAvailabilityManager` contract will query the corresponding vote aggregation storage to verify if enough votes have been registered. Upon the third vote a transaction to `RewardsManager` is triggered to make the subgraph rewards state change.
+
+A `voteTimeLimit` parameter will be used to invalidate old votes, initially set to ten minutes. This duration is slightly longer than the current subgraph availability oracle configuration, which runs every five minutes, allowing time for oracles to cast their votes and accounting for potential delays in execution. This minimizes the need for frequent transactions. When comparing oracle votes, the `SubgraphAvailabilityManager` will use the `voteTimeLimit` to ensure the last recorded vote occurred within this timeframe. Votes older than the limit will not be considered valid, necessitating a new vote from the oracle if required. It's crucial to note that any adjustments to `voteTimeLimit` must always exceed the oracle's execution period to prevent the premature invalidation of votes before all oracles have participated. However, this timeframe should be kept as brief as possible, since an excessively long `voteTimeLimit` undermines its intended purpose. The council can modify this timeframe if desired but it must be set with consideration to the oracle’s iteration period to avoid repeated voting or expired votes.
+
+# Rationale and Alternatives
+
+The presented proposal serves as a temporary solution; we will continue our efforts to explore additional methods for further decentralization or even the complete elimination of the subgraph availability oracle. We believe this implementation provides a short-term fix that significantly improves decentralization compared to the previous 1-of-1 approach.
+
+---
+
+*This section contains the body of the Subgraph Availability Oracle Operator Charter.*
+
+# Subgraph Availability Oracle Operator Charter
+
+## 1. Role of the Subgraph Availability Oracle.
+
+The primary role of the Subgraph Availability Oracle in The Graph Network is to ensure the availability and validity of subgraph manifests. Oracles are responsible for verifying subgraphs availability and conducting validity checks.
+
+Oracles operate as part of the `SubgraphAvailabilityManager` smart contract, contributing to the decentralized decision-making process by casting votes regarding the availability and validity of subgraph files.
+
+## 2. Appointment and Governance of Subgraph Availability Oracles Operators
+
+Oracle operators are appointed by The Graph Council, which retains the authority to add or remove oracle operators, ensuring compliance with the guidelines and responsibilities outlined in this charter.
```

Comment:
Is there any incentives for these operators to maintain integrity and high availability?

Reply:
Oracles will be run initially by core devs so there's already incentives I believe 🙂

Reply:
sounds reasonable, I surely hope so:) 
would you include this assumption in the GIP and perhaps also alternative considerations when oracles aren't run by core devs, if we ever expect this to happen? I can also understand if you consider this to be out of scope

Reply:
I'd upvote @hopeyen's proposal to include this info in the GIP. I had the exact same question reading through.

Reply:
Sounds good, I'll include in the GIP that on this first iteration oracles will be run by core developer teams.